tests/integration/main.rs

```diff
@@ -248,3 +253,146 @@ fn test_update_property() {
     assert_eq!(&inserted.key, &fetched.key);
     assert_eq!(&new_string_prop, &fetched.prop_string);
 }
+
+#[test]
+#[cfg_attr(not(feature = "integration_tests"), ignore)]
+fn test_get_by_array_property() {
+    let connection = create_test_connection();
+
+    // Generate some test entities
+    let string_value_a = generate_random_string(10);
+    let string_value_b = generate_random_string(10);
+    let string_value_c = generate_random_string(10);
+
+    let mut entity_a = generate_random_entity();
+    entity_a.prop_string_array = vec![string_value_a.clone(), string_value_b.clone()];
+    let mut entity_b = generate_random_entity();
+    entity_b.prop_string_array = vec![string_value_b.clone(), string_value_c.clone()];
+
+    // Insert
+    let inserted_a = match entity_a.commit(&connection) {
```

Comment:
You can seriously improve your test code if you follow this advice:

```
In Rust 2018 you can instead let your #[test]s and main functions return a Result:


// Rust 2018

use std::fs::File;

fn main() -> Result<(), std::io::Error> {
    let f = File::open("bar.txt")?;

    Ok(())
}
```

Reply:
yes, I saw that yesterday, will fix that , as you say that could remove a lot of code and increase readability